common/src/main/java/com/wynntils/features/DiscordRichPresenceFeature.java

```diff
@@ -90,6 +94,15 @@ public void onWorldChange(WorldStateEvent event) {
         }
     }
 
+    @SubscribeEvent
+    public void onStreamToggle(StreamModeEvent e) {
+        if (disableInStream.get() && e.isEnabled()) {
+            onDisable();
+        } else if (!e.isEnabled()) {
+            onEnable();
+        }
```

Comment:
Can you extract what onEnable/onDIsable does into enable/disable methods, and call that here and in the onX methods? It feels very wrong to call such a feature "event" method

Reply:
Also, you may need to add hooks for "config change" here.